ext/curl/tests/gh16802.phpt

```diff
@@ -0,0 +1,32 @@
+--TEST--
+GH-16802 (open_basedir bypass using curl extension)
+--EXTENSIONS--
+curl
+--SKIPIF--
+<?php
+if (PHP_OS_FAMILY === "Windows") die("skip not for Windows");
```

Comment:
Why not? The test passes for me (checked `master` only, though).

Reply:
Because the file paths are Linux file paths. It passes because the check prevents the access anyway, but I didn't want to deal with Windows and check what happens if the patch is not applied. I can remove this check if you want.

Reply:
Without the patch, only the fourth `curl_setopt()` would trigger a warning, and `curl_exec()` would return `false` (with `CURLOPT_VERBOSE` "Protocol "file" disabled" is reported). So the actual file:// URI wouldn't matter. Doing `curl_exec()` after `each curl_setopt()` would only fail due to "Couldn't open file /etc/passwd" for the first exec; the three following `curl_exec()` would fail due to "Protocol "file" disabled" (still without the patch).

Anyhow, since the `open_basedir` value doesn't matter, we *could* use `"file://" . str_replace("\\", "/", __FILE__)` instead of `file://etc/passwd`, but it seems to me checking for the warnings is good enough. Even the `curl_exec()` is not really relevant.